omnigibson/envs/env_base.py

```diff
@@ -80,6 +93,21 @@ def __init__(
         # Load this environment
         self.load()
 
+        # Set the initial camera if required.
+        self.use_external_camera = use_external_camera
+        if use_external_camera:
+            self.camera = VisionSensor(
+                prim_path="/World/viewer_camera",
+                name="camera",
+                modalities=external_camera_modalities, 
+            )
+            self.camera.initialize()
```

Comment:
I think the camera has to be loaded first, right (`cam.load()`)? (i.e.: the prim needs to get created) I think this is why it didn't work when you tried to create a separate camera.